postupgrade/mergeGlobalConfigs.js

```diff
@@ -0,0 +1,174 @@
+  const incomingJson = parse(incomingConfig);
+  const originalTokens = tokenize(originalJson);
+  const incomingTokens = tokenize(incomingJson);
+  const mergedTokens = mergeTokens(originalTokens, incomingTokens);
+  const commentJsonObject = transformToCommentJsonObject(mergedTokens);
+  if (incomingJson.sdkVersion) {
+    commentJsonObject.sdkVersion = incomingJson.sdkVersion
+  }
+  return stringify(commentJsonObject, null, 2);
+}
+
+/**
+ * Parses a {@link CommentJSONObject} into an array of {@link GlobalConfigToken}s
+ * 
+ * @param {CommentJSONObject}  
+ * @returns {GlobalConfigToken[]}
+ */
+function tokenize(commentJsonObject) {
+  /** @type {GlobalConfigToken[]} */
+  const tokens = []
+
+  /**
+   * @param {CommentToken} comment 
+   * @returns {CommentedOutPropertyToken}
+   */
+  const parseCommentedOutProperty = (comment) => {
+    const propCommentRegex = /^\s?"([^"]+)":.*$/g
+    const val = comment.value
+    const match = [...val.matchAll(propCommentRegex)][0]
+    if (!match || match.length !== 2) {
+      throw new Error(`Could not parse global config commented out property: \`${val}\``)
+    }
+    return {
+      type: 'CommentedOutProperty',
+      key: match[1],
+      comment: {
+        type: comment.type,
+        value: comment.value,
+        inline: comment.inline
+      }
+    }
+  }
+
+  const handleComment = c => {
+    if (c.inline) return;
+    tokens.push(parseCommentedOutProperty(c));
+  }
+
+  Object.keys(commentJsonObject).forEach(key => {
+    const beforeSymbol = Symbol.for(`before:${key}`)
+    const afterSymbol = Symbol.for(`after:${key}`)
+    const beforeComments = commentJsonObject[beforeSymbol] || []
+    const afterComments = commentJsonObject[afterSymbol] || []
+    beforeComments.forEach(handleComment)
+    const inlineComment = afterComments.find(c => c.inline);
+    tokens.push({
+      type: 'Property',
+      key,
+      value: commentJsonObject[key],
+      inlineComment: inlineComment ? {
+        type: inlineComment.type,
+        value: inlineComment.value,
+        inline: inlineComment.inline
+      } : null
+    })
+    afterComments.forEach(handleComment)
+  })
+
+  return tokens;
+}
+exports.tokenize = tokenize;
+
+/**
+ * Transforms an array of {@link GlobalConfigToken} into an equivalent {@link CommentJSONObject}
+ * 
+ * @param {GlobalConfigToken[]} tokens 
+ * @returns {CommentJSONObject}
+ */
+function transformToCommentJsonObject(tokens) {
+  const commentJsonObject = {}
+  let previousPropertyKey = null;
+  tokens.forEach(token => {
+    switch (token.type) {
+      case 'Property':
+        commentJsonObject[token.key] = token.value
+        previousPropertyKey = token.key
+        if (token.inlineComment) {
+          const symbol = Symbol.for(`after:${token.key}`);
+          commentJsonObject[symbol] = (commentJsonObject[symbol] || []).concat([token.inlineComment])
+        }
+        break;
+      case 'CommentedOutProperty':
```

Comment:
Do we handle the case of a line that's a `CommentedOutProperty` that also has an `inline-comment`?

Reply:
`CommentedOutProperty`s  don't know about whether they contain inline comments or not. All they know is that they have some string value. That string value might have what we recognize as an inline comment inside of it, but, currently, `CommentedOutProperty`s don't need to know about that. I don't think we would need to make large changes to add that information later on.

Reply:
tldr is yes it's handled correctly, since it's not a case we need to do anything about